cameraServer.py

```diff
@@ -0,0 +1,215 @@
+
+
+
+#Process Functions
+
+def getRetroPos(img, display=False, sample=False):
+    """Function for finding retro-reflective tape a"""
+    hsv = cv2.cvtColor(img, cv2.COLOR_BGR2HSV) #Convert to HSV to make the mask easier
+    if sample:
+        mask = cv2.inRange(hsv, sample_lowerGreen, sample_higherGreen) #Create a mask of everything in between the greens
+    else:
+        mask = cv2.inRange(hsv, lowerGreen, higherGreen)
+    mask = cv2.dilate(mask, None, iterations=2) #Expand the mask to allow for further away tape
+
+    contours = cv2.findContours(mask, 1, 2)[-2] #Find the contours
+    
+    if len(contours) > 1: #Get contours with area above magic number 10 and append its smallest rectangle
+        rects = []
+        for cnt in contours:
+            if cv2.contourArea(cnt) > minContourArea:
+                rects.append(cv2.minAreaRect(cnt))
+        rects.sort(key=lambda x:x[2])
+        
+        pairs = []
+        leftRect = None
+        for i, rect in enumerate(sorted(rects, key=lambda x:x[0])): #Get rectangle pairs
+            if rect[2] < (leftAngleSize + angleOffset) and rect[2] > (leftAngleSize - angleOffset):
+                leftRect = rect
+            elif rect[2] < (rightAngleSize + angleOffset) and rect[2] > (rightAngleSize - angleOffset):
+                if leftRect:
+                    pairs.append((leftRect, rect))
+                    leftRect = None
+                    
+        if len(pairs) >= 1:
+            closestToMiddle = min(pairs, key = lambda x:abs(x[0][0][0] - screenSize[0]/2))
+        else:
+            return None, None, img
+
+        circleContours = list(np.int0(cv2.boxPoints(closestToMiddle[0])))
+        circleContours.extend(list(np.int0(cv2.boxPoints(closestToMiddle[1]))))
+        circleContours = np.array(circleContours)
+        (x,y),radius = cv2.minEnclosingCircle(circleContours)
+        radius = int(radius)
+        center = (int(x), int(y))
+
+        angle = (closestToMiddle[0][1][0] * closestToMiddle[0][1][1]) / (closestToMiddle[1][1][0] * closestToMiddle[1][1][1])
+        
+        if display == True: #Create the annotated display if display is True
+            for pair in pairs:
+                if pair == closestToMiddle:
+                    for tape in pair:
+                        img = cv2.drawContours(img, [np.int0(cv2.boxPoints(tape))], 0, (0, 255, 0), thickness=2)
+                    img = cv2.circle(img, center, radius, (0, 255, 0))
+                else:
+                    for tape in pair:
+                        img = cv2.drawContours(img, [np.int0(cv2.boxPoints(tape))], 0, (0, 0, 255))
+
+            return angle, -(((x/screenSize[0])*2)-1), img
+    return None, None, img
+
+def getGroundPos(img, sample=False): #TODO: Ground Tape Function
+    return None, None, None #Distance from x, distance from y, angle of line
+    
+#if False:
+if __name__ == "__main__":
+    from cscore import CameraServer, VideoSource
+    from networktables import NetworkTables
```

Comment:
Read [pep 8](https://pep8.org/#imports) for how to structure imports, hint, should go at the top of the file